examples/pic.cpp

```diff
@@ -0,0 +1,363 @@
+        stream << u.pos << u.val;
+        return stream;
+    }
+    friend husky::BinStream& operator>>(husky::BinStream& stream, MatElem& u) {
+        stream >> u.pos >> u.val;
+        return stream;
+    }
+
+    KeyT pos;
+    float val;
+};
+
+class VElem {
+   //Element for the classification vector V
+   public:
+    //using KeyT = std::pair<uint, uint>; // std::hash don't like std::pair
+    using KeyT = int;
+
+    VElem() {}
+    explicit VElem(const KeyT& p) : pos(p), val(0.), valp(0.), valpp(0.) {}
+    virtual const KeyT& id() const { return pos; }
+
+    // Serialization and deserialization
+    friend husky::BinStream& operator<<(husky::BinStream& stream, VElem u) {
+        stream << u.pos << u.val << u.valp << u.valpp;
+        return stream;
+    }
+    friend husky::BinStream& operator>>(husky::BinStream& stream, VElem& u) {
+        stream >> u.pos >> u.val >> u.valp >> u.valpp;
+        return stream;
+    }
+
+    KeyT pos;
+    float val;
+    float valp;  //val at previous iteration
+    float valpp; //val at previousX2 iteration
+
+    void setVal(float newval){
+        valpp = valp;
+        valp  = val;
+        val   = newval;
+    }
+
+    float getAcc(){
+        return fabs(fabs(val-valp) - fabs(valp-valpp));
+    }
+};
+
+class DimStat {
+   //Dimension statistics
+   public:
+    using KeyT = int;
+
+    DimStat() {}
+    explicit DimStat(const KeyT& w) : dimId(w) {}
+    virtual const KeyT& id() const { return dimId; }
+
+    // Serialization and deserialization
+    friend husky::BinStream& operator<<(husky::BinStream& stream, DimStat u) {
+        stream << u.dimId << u.mean << u.var;
+        return stream;
+    }
+    friend husky::BinStream& operator>>(husky::BinStream& stream, DimStat& u) {
+        stream >> u.dimId >> u.mean >> u.var;
+        return stream;
+    }
+
+    int dimId;
+    float mean;
+    float var;
+};
+
+
+void pic() {
+    husky::io::HDFSLineInputFormat infmt;
+    infmt.set_input(husky::Context::get_param("input"));
+
+    int dimension = std::stoi(husky::Context::get_param("dimension"));
+
+    // 1. Create and globalize ndpoint objects
+    husky::ObjList<NDpoint> ndpoint_list;
+    auto parse_ndpoint = [&ndpoint_list](boost::string_ref& chunk) {
+        if (chunk.size() == 0)
+            return;
+        boost::char_separator<char> sep(" \t");
+        boost::tokenizer<boost::char_separator<char>> tok(chunk, sep);
+        boost::tokenizer<boost::char_separator<char>>::iterator it = tok.begin();
+        int id = stoi(*it++);
+        std::vector<float> coords;
+        while (it != tok.end()) {
+            coords.push_back(stod(*it++));
+        }
+        ndpoint_list.add_object(NDpoint(id, coords));
+    };
+    load(infmt, parse_ndpoint);
+    globalize(ndpoint_list);
+    husky::base::log_msg( "ndpoint_list size " + std::to_string( ndpoint_list.get_size()));
+
+    husky::base::log_msg( "part1 ok ");
```

Comment:
I wonder whether it's possible to make the logs more meaningful? E.g., something like "Data points loaded". And I guess they will flood the screen if you use many threads?